packages/compass-collection/src/stores/index.ts

```diff
@@ -1,395 +1,129 @@
-    }
   });
 
-  /**
-   * Remove any open tabs when database dropped.
-   *
-   * @param {String} name - The name.
-   */
-  appRegistry.on('database-dropped', (name: string) => {
-    store.dispatch(databaseDropped(name));
-
-    const currentNamespace = store.getState().namespace;
-    if (currentNamespace) {
-      const { database } = toNS(currentNamespace as string);
-      if (name === database) {
-        appRegistry.emit('active-database-dropped', name);
-      }
-    }
+  globalAppRegistry.on('database-dropped', (namespace: string) => {
+    store.dispatch(databaseDropped(namespace));
   });
 
   /**
    * Set the data service in the store when connected.
    */
-  appRegistry.on(
+  globalAppRegistry.on(
     'data-service-connected',
     (error, dataService: DataService) => {
-      store.dispatch(dataServiceConnected(error, dataService));
+      store.dispatch(dataServiceConnected());
+      thunkExtraArg.dataService = dataService;
     }
   );
 
   /**
    * When we disconnect from the instance, clear all the tabs.
    */
-  appRegistry.on('data-service-disconnected', () => {
-    store.dispatch(clearTabs());
+  globalAppRegistry.on('data-service-disconnected', () => {
+    store.dispatch(dataServiceDisconnected());
+    thunkExtraArg.dataService = null;
   });
 
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  const ipc = require('hadron-ipc');
+
   // TODO: importing hadron-ipc in unit tests doesn't work right now
-  if (ipc.on) {
-    /**
-     * When `Share Schema as JSON` clicked in menu send event to the active tab.
-     */
+  if (ipc?.on) {
     ipc.on('window:menu-share-schema-json', () => {
-      const state = store.getState();
-      if (state.tabs) {
-        const activeTab = state.tabs.find(
-          (tab: WorkspaceTabObject) => tab.isActive === true
-        );
-        if (activeTab.localAppRegistry) {
-          activeTab.localAppRegistry.emit('menu-share-schema-json');
-        }
+      const activeTab = getActiveTab(store.getState());
+
+      if (!activeTab) {
+        return;
       }
+
+      activeTab.localAppRegistry.emit('menu-share-schema-json');
     });
 
     ipc.on('compass:open-export', () => {
-      const state = store.getState();
-      if (!state.tabs) {
-        return;
-      }
-      const activeTab = state.tabs.find(
-        (tab: WorkspaceTabObject) => tab.isActive === true
-      );
+      const activeTab = getActiveTab(store.getState());
+
       if (!activeTab) {
         return;
       }
-      const crudStore = activeTab.localAppRegistry.getStore('CRUD.Store');
-      const { query: crudQuery, count } = crudStore.state;
 
-      const { filter, project, collation, limit, skip, sort } = crudQuery;
-      appRegistry.emit('open-export', {
+      globalAppRegistry.emit('open-export', {
         exportFullCollection: true,
         namespace: activeTab.namespace,
-        query: { filter, project, collation, limit, skip, sort },
```

Comment:
`exportFullCollection` is always `true` (this is existing behavior) and removing this allow to remove a dependency on crud store here